packages/compass-aggregations/src/modules/max-time-ms.ts

```diff
@@ -6,18 +6,53 @@ import { isAction } from '../utils/is-action';
 
 export const MAX_TIME_MS_CHANGED =
   'aggregations/max-time-ms/MAX_TIME_MS_CHANGED' as const;
+export const PREFERENCES_MAX_TIME_MS_CHANGED =
+  'aggregations/max-time-ms/PREFERENCES_MAX_TIME_MS_CHANGED' as const;
 export interface MaxTimeMSChangedAction {
   type: typeof MAX_TIME_MS_CHANGED;
   maxTimeMS: number | null;
 }
+export interface PreferencesMaxTimeMSChangedAction {
+  type: typeof PREFERENCES_MAX_TIME_MS_CHANGED;
+  maxTimeMS: number | null;
+}
 
-type State = null | number;
+type State = {
+  current: number | null;
+  preferencesValue: number | null;
+};
 
-export const INITIAL_STATE: State = null;
+export const INITIAL_STATE: State = { current: null, preferencesValue: null };
```

Comment:
I think I wouldn't expect that we need to store the preferences maxTimeMs in the store and sync it manually, we can read it from the injected service in actions (we only care about the current value for running queries and aggs, right?) and access using usePreference hooks in the UI where needed. I might be missing something here though so wondering why you decided to go that way? 🤔

Reply:
Yeah, I tried to do that at first, and it's definitely possible. But when I started to actually make that change, the number of places where this is being used turned out to be quite high, so I kept it this way to keep the change a bit more "localized" and avoid the risk of missing a necessary maxTimeMS check in one place

(plus, the maxTimeMS state *is* synced to UI state after all, so it's not just something that gets applied on top of other state in thr plugin)

Reply:
No strong preference if you want to keep it that way, it's just one preference value in this case after all, but generally speaking we should aim for redux stores not holding a copy of the state that already exists elsewhere, otherwise it's hard to navigate state dependencies: the old way of handling namespace state or a various copied over instance model props is a good example how it can become a mess to maintain quickly.